staging/src/k8s.io/client-go/transport/config.go

```diff
@@ -68,6 +69,12 @@ type Config struct {
 
 	// Dial specifies the dial function for creating unencrypted TCP connections.
 	Dial func(ctx context.Context, network, address string) (net.Conn, error)
+
+	// Proxier is the the proxier to be used for all requests made by this
+	// transport.
+	//
+	// socks5 proxying does not currently support spdy streaming endpoints.
+	Proxier func(*http.Request) (*url.URL, error)
```

Comment:
same comment about naming it `Proxy` and defining behavior when nil, and when returning `nil, nil`

Reply:
Done.